--- origsrc/sqlite-3.6.3/Makefile.in	2008-09-26 03:16:33.181896000 -0500
+++ src/sqlite-3.6.3/Makefile.in	2008-09-26 03:16:50.447521000 -0500
@@ -747,7 +747,7 @@
 	$(INSTALL) -m 0644 sqlite3.pc $(DESTDIR)$(pkgconfigdir)
 
 tcl_install:	libtclsqlite3.la
-	$(TCLSH_CMD) $(TOP)/tclinstaller.tcl $(VERSION)
+	$(TCLSH_CMD) $(TOP)/tclinstaller.tcl $(RELEASE)
 
 clean:	
 	rm -f *.lo *.la *.o sqlite3$(TEXE) libsqlite3.la
--- origsrc/sqlite-3.6.3/src/tclsqlite.c	2008-09-09 07:31:34.000000000 -0500
+++ src/sqlite-3.6.3/src/tclsqlite.c	2008-09-26 03:17:08.978771000 -0500
@@ -1343,11 +1343,11 @@
       Tcl_AppendResult(interp,"Error: non-null separator required for copy",0);
       return TCL_ERROR;
     }
-    if(sqlite3StrICmp(zConflict, "rollback") != 0 &&
-       sqlite3StrICmp(zConflict, "abort"   ) != 0 &&
-       sqlite3StrICmp(zConflict, "fail"    ) != 0 &&
-       sqlite3StrICmp(zConflict, "ignore"  ) != 0 &&
-       sqlite3StrICmp(zConflict, "replace" ) != 0 ) {
+    if(strcmp(zConflict, "rollback") != 0 &&
+       strcmp(zConflict, "abort"   ) != 0 &&
+       strcmp(zConflict, "fail"    ) != 0 &&
+       strcmp(zConflict, "ignore"  ) != 0 &&
+       strcmp(zConflict, "replace" ) != 0 ) {
       Tcl_AppendResult(interp, "Error: \"", zConflict, 
             "\", conflict-algorithm must be one of: rollback, "
             "abort, fail, ignore, or replace", 0);
