--- origsrc/sqlite-3.5.9/Makefile.in	2008-05-12 12:10:26.000000000 -0500
+++ src/sqlite-3.5.9/Makefile.in	2008-05-21 23:35:02.078125000 -0500
@@ -413,13 +413,14 @@
 
 libsqlite3.la:	$(LIBOBJ)
 	$(LTLINK) -o $@ $(LIBOBJ) $(TLIBS) \
-		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
+		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8" -no-undefined
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -o $@ tclsqlite.lo \
-		$(LIBOBJ) @TCL_STUB_LIB_SPEC@ $(TLIBS) \
+		libsqlite3.la $(LIBTCL) $(LIBPTHREAD) \
                 -rpath "$(libdir)/sqlite" \
-		-version-info "8:6:8"
+		-avoid-version -no-undefined
+
 
 sqlite3$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
 	$(LTLINK) $(READLINE_FLAGS) \
--- origsrc/sqlite-3.5.9/tclinstaller.tcl	2008-03-14 13:01:53.000000000 -0500
+++ src/sqlite-3.5.9/tclinstaller.tcl	2008-05-21 23:35:40.343750000 -0500
@@ -5,7 +5,7 @@
 #    tclsh tclinstaller.tcl 3.0
 #
 set VERSION [lindex $argv 0]
-set LIBFILE .libs/libtclsqlite3[info sharedlibextension]
+set LIBFILE .libs/cygtclsqlite3[info sharedlibextension]
 if { ![info exists env(DESTDIR)] } { set env(DESTDIR) "" }
 if { ![info exists env(TCLLIBDIR)] } { set env(TCLLIBDIR) [lindex $auto_path 0] }
 set LIBDIR $env(DESTDIR)$env(TCLLIBDIR)
